### PR TITLE
[CRFLN-1] Set up the React project

### PR DESCRIPTION
**Pull request by Fahleen1**

### CODE REVIEW

File: currency-converter/src/App.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  }
+}
+
+.card {
+  padding: 2em;
+}
+
+.read-the-docs {
+  color: #888;
+} */
```
Comment (riazosama) [inline currency-converter/src/App.css]: We should never commit commented code unless there is no other way. In that case, we should add a comment stating why do we need the commented code.

File: currency-converter/src/components/MainHeading.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export default function MainHeading({name}:{name:string}) {
+  return (
+    <>
```
Comment (riazosama) [inline currency-converter/src/components/MainHeading.tsx]: We do not need to add a fragment here. A fragment is just a `div`.

File: currency-converter/src/index.css [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    color: #213547;
+    background-color: #ffffff;
+  }
+  a:hover {
+    color: #747bff;
+  }
+  button {
+    background-color: #f9f9f9;
+  }
+} */
```
Comment (riazosama) [inline currency-converter/src/index.css]: same as above

**Review by riazosama — CHANGES_REQUESTED** [review]
you forgot to add prettier config

File: currency-converter/eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import js from '@eslint/js'
+import globals from 'globals'
+import reactHooks from 'eslint-plugin-react-hooks'
+import reactRefresh from 'eslint-plugin-react-refresh'
```
Comment (riazosama) [inline currency-converter/eslint.config.js]: what's the use of this? I've not seen this used hence asking.

File: currency-converter/eslint.config.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import js from '@eslint/js'
```
Comment (riazosama) [inline currency-converter/eslint.config.js]: let's rename this to `eslintJS`. it more inline with eslint than simple `js`

File: currency-converter/tsconfig.node.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+    /* Linting */
+    "strict": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noFallthroughCasesInSwitch": true,
+    "noUncheckedSideEffectImports": true
+  },
+  "include": ["vite.config.ts"]
+}
```
Comment (riazosama) [inline currency-converter/tsconfig.node.json:24]: do you know what's these are doing? maybe do a quick chatgpt and explain this to me?

**Review by riazosama — CHANGES_REQUESTED** [review]

File: currency-converter/tsconfig.node.json (hunk 6 above)
Comment (Fahleen1) [reply currency-converter/tsconfig.node.json:24]: It's defining how typescript will use compiler the first part specifies location to store build information, sets JS version, includes built-in libraries, tells how modules should be generated.
Bundler mode specifies compatibility with bundlers like vite
Linting enables strict type checking, ensures no unused variable and parameter is used.

**Review by riazosama — APPROVED** [review]

File: currency-converter/eslint.config.js (hunk 4 above)
Comment (Fahleen1) [reply currency-converter/eslint.config.js]: These tools are use to identify and resolve problems Typescript code and globals are used to add package globally , react hooks will help in importing hooks while react refresh provides updates for react components.

**Pull request merged**